crates/bevy_render/src/render_resource/buffer_vec.rs

```diff
@@ -3,13 +3,15 @@ use crate::{
     renderer::{RenderDevice, RenderQueue},
 };
 use bevy_core::{cast_slice, Pod};
+use std::ops::{Deref, DerefMut};
 use wgpu::BufferUsages;
 
+/// A user-friendly wrapper around a [`Buffer`] that provides a `Vec`-like
+/// interface for constructing the buffer.
 pub struct BufferVec<T: Pod> {
     values: Vec<T>,
```

Comment:
```suggestion
    pub values: Vec<T>,
```

If we're allowing `DerefMut` access to this, we should also make this field pub for consistency.

Reply:
I don't think we should. If for whatever reason we want to move the Vec out of the BufferVec/Uniform Vec, I'd prefer a `take`-like API that consumes the wrapper to return the wrapped Vec. This assures the Buffer is dropped properly and the BufferVec is not left in a partially consumed state.

Reply:
Ah, that's a good argument 👍🏽